### PR TITLE
Implement vertical gridlist

### PR DESCRIPTION
**Pull request by NiGhTTraX**
# Vertical gridlist

In order to make the grid to support both `vertical` and `horizontal` directions we will modify its current implementation by adding new concepts `#indexes` and `#lanes`.

**Insert the image** 

- [x] Modify `rows` prop to `itemsPerLane`, we no longer need a `rows` option
*If specified, `itemsPerLane`, will represent the #lanes*
- [x] Pass a custom option `direction` to identify the direction of the grid. 
*By default will be `horizontal`. `direction` can be `horizontal` (default) or `vertical`*
- [x] Fix failing unit tests
- [x] Add the fixtures for vertical grid as presented below

#### Item per lane 2

Image
2_items_per_lane

Fixture

```javascript
itemsPerLane2: [
  {w: 1, h: 1, x: 0, y: 0},
  {w: 1, h: 1, x: 1, y: 0},
  {w: 1, h: 2, x: 0, y: 1},
  {w: 1, h: 1, x: 1, y: 1},
  {w: 1, h: 2, x: 1, y: 2},
  {w: 1, h: 1, x: 0, y: 3},
  {w: 0, h: 1, x: 0, y: 4},
  {w: 1, h: 1, x: 0, y: 5},
  {w: 1, h: 1, x: 1, y: 5},
  {w: 2, h: 2, x: 0, y: 6},
  {w: 2, h: 1, x: 0, y: 8},
]
```

Resize flow from `2 lanes => 3 lanes` and from `3 lanes => 4 lanes`
image2

#### Item per lane 3

Image
3_items_per_lane

Fixture

```javascript
itemsPerLane3: [
  {w: 1, h: 1, x: 0, y: 0},
  {w: 1, h: 1, x: 1, y: 0},
  {w: 1, h: 1, x: 2, y: 0},
  {w: 2, h: 1, x: 0, y: 1},
  {w: 1, h: 1, x: 2, y: 1},
  {w: 0, h: 1, x: 0, y: 2},
  {w: 1, h: 2, x: 0, y: 3},
  {w: 1, h: 1, x: 1, y: 3},
  {w: 1, h: 1, x: 2, y: 3},
  {w: 1, h: 1, x: 1, y: 4},
  {w: 1, h: 1, x: 2, y: 4},
  {w: 2, h: 1, x: 0, y: 5},
  {w: 1, h: 2, x: 2, y: 5},
  {w: 1, h: 1, x: 0, y: 6},
  {w: 1, h: 1, x: 1, y: 6},
  {w: 2, h: 2, x: 0, y: 7},
  {w: 1, h: 2, x: 2, y: 7}
]
```

Resize flow from `3 lanes => 2 lanes` and from `3 lanes => 4 lanes`

image3

#### Item per lane 4

Image
4_items_per_lane

Fixture

```javascript
itemsPerLane4: [
  {w: 2, h: 1, x: 0, y: 0},
  {w: 1, h: 1, x: 2, y: 0},
  {w: 1, h: 2, x: 3, y: 0},
  {w: 2, h: 2, x: 0, y: 1},
  {w: 1, h: 2, x: 2, y: 1},
  {w: 1, h: 1, x: 3, y: 2},
  {w: 0, h: 1, x: 0, y: 3},
  {w: 2, h: 1, x: 0, y: 4},
  {w: 1, h: 1, x: 2, y: 4},
  {w: 1, h: 1, x: 3, y: 4},
  {w: 1, h: 2, x: 0, y: 5},
  {w: 1, h: 1, x: 1, y: 5},
  {w: 2, h: 1, x: 2, y: 5},
  {w: 2, h: 1, x: 1, y: 6},
  {w: 1, h: 1, x: 3, y: 6}
]
```

Resize flow from `4 lanes => 5 lanes` and from `4 lanes => 3 lanes`

image4

----

- [x] Test that items are correctly positioned withing the grid when `direction = vertical`
- [x] Write the code which will position the items for `vertical` direction, ensure that unit test passed.

- [ ] Further Todos
  - [ ] tests
    - [ ] Write tests for collisions
    - [ ] Write tests for item resize
    - [ ] Write tests for grid changes
    - [ ] Write tests for `horizontal "timelines"`
  - [ ] implementation
    - [ ] adapt `_tryToResolveCollisionsLocally` and `_pullItemsToLeft` for vertical grid
    - [ ] refactor `_adjustHeightOfItems` to both directions

------

**#lanes** are `rows` or `columns` based on the `direction`. Lanes represent the fixed dimension of the grid.

**#indexes** = represent the variable dimension of the grid. 

`grid[index][lane]`

### CODE REVIEW

**Comment by andrei-picus-hs** [comment]
@skidding care to take a look?

File: README.md [hunk 1]
```diff
@@ -212,10 +216,7 @@ divided by the number of grid rows.
 
 - Their README reads Ducksboard is no longer active in their development. There
 are a few notable forks but it's hard to assert their [reliability.](https://github.com/dustmoo/gridster.js/issues)
-- gridster works vertically and our design is horizontal. We instigated a
```
Comment (ovidiu-chereches-hs) [inline README.md]: I think it's still worth noting that GridList (now) supports both vertical and horizontal, while Gridster just vertical. Unless it now supports both axis.

**Comment by ovidiu-chereches-hs** [comment]
LGTM

File: demo/style.css [hunk 2]
```diff
@@ -12,8 +12,6 @@ body {
   left: 0;
   right: 10px;
   bottom: 0;
-  overflow-x: auto;
-  overflow-y: hidden;
```
Comment (catalinmiron) [inline demo/style.css]: You can add `auto` for both `x` and `y`

**Comment by catalinmiron** [comment]
@NiGhTTraX I found a collision bug or is this the desired behaviour ?
grid_collisions_bug

**Comment by skidding** [comment]
:+1:

**Comment by andrei-picus-hs** [comment]
@catalinmiron fixed.

**Comment by catalinmiron** [comment]
@andrei-picus-hs :star2: LGTM
grid_collisions_are_working

**Pull request merged**